### PR TITLE
T2 spatial interaction

### PR DESCRIPTION
**Pull request by ruiheesi**
HI George, this is the update for moving some functions from NDIAP to local, including:
1. Set default color map for heatmap to seismic for better visual representation
2. Add the pandas dataframe with proper column name and row index into the return:
    a). Add function to extract the labels from the figure object
    b). Add function to process the matrix to assign proper row index and column name
    c). Return the proper file name to support image data export as well.
3. Update the docstring of the function

Thanks. The JIRA ticket link: https://tracker.nci.nih.gov/browse/DSP-455

### CODE REVIEW

File: src/spac/spatial_analysis.py [hunk 1]
```diff
@@ -90,23 +91,51 @@ def spatial_interaction(
...
+            returns a dictionary of Axes objects, with keys
+            representing the stratification groups.
+
+        Matrix : dict or tuple, optional
+            Returned only if `return_matrix=True`. If `stratify_by` is
+            not used, returns:
+            - For "Neighborhood Enrichment": a tuple with the z-score
+              matrix and enrichment counts matrix.
+            - For "Cluster Interaction Matrix": the interaction
+              matrix. If `stratify_by` is used, returns a dictionary
```
Comment (georgezakinih) [inline src/spac/spatial_analysis.py:125]: @ruiheesi, line 120 mentions that stratify_bis is not used, but this line says it is used. Does the general condition that stratify_by is not used applies to this line 125?

File: src/spac/spatial_analysis.py [hunk 2]
```diff
@@ -90,23 +91,51 @@ def spatial_interaction(
...
+              of matrices, with keys representing the stratification
+              groups.
+
+        Table : dict, optional
+            Returned only if `return_matrix=True`. Contains processed
+            DataFrames of computed matrices with row and column labels
+            applied. If `stratify_by` is used, the keys represent the
```
Comment (georgezakinih) [inline src/spac/spatial_analysis.py:132]: @ruiheesi , I am not clear why we have Matrix and Table. Looks that Table will include the information that are in Matrix.

File: src/spac/spatial_analysis.py [hunk 3]
```diff
@@ -251,6 +280,79 @@ def perform_analysis(
 
         return ax
 
+    def get_labels(
```
Comment (georgezakinih) [inline src/spac/spatial_analysis.py:283]: @ruiheesi , please add a doc string for this function

Comment (ruiheesi) [reply src/spac/spatial_analysis.py]: Hi George, I have addressed in the new commit, I will close this conversation and please review the latest content. Thanks

File: src/spac/spatial_analysis.py [hunk 4]
```diff
@@ -251,6 +280,79 @@ def perform_analysis(
...
+                        print(
+                            f"Column labels found on axis {i}: {xticklabels}"
+                        )
+                    row_labels = xticklabels[::-1]
+
+        return row_labels
+
+    # Use to process the output from different
+    # spatial analysis method in squidpy
+    def process_matrixs(matrixs, row_labels, plot_label=None):
```
[the remark this answers is not in the document]
Comment (ruiheesi) [reply src/spac/spatial_analysis.py]: Hi George, I have addressed in the new commit, I will close this conversation and please review the latest content. Thanks

File: src/spac/spatial_analysis.py (hunk 2 above)
Comment (ruiheesi) [reply src/spac/spatial_analysis.py]: Hi George, I have addressed in the new commit, I will close this conversation and please review the latest content. Thanks

**Pull request merged**